### ISSUE TITLE
Bunch of Errors starting a new app

### ISSUE THREAD

**Issue by folivi** [wix/react-native-controllers#85]
Hi guys,

I just started a new app, imported reac-native-controllers and the run into a bunch of errors

node version 6.3.1 (happens with 5.10 too)

my dependencies

```
"dependencies": {
    "react": "15.2.1",
    "react-native": "^0.31.0",
    "react-native-fbsdk": "^0.3.0",
    "react-native-navigation": "^1.0.30",
    "react-native-vector-icons": "^2.0.3",
    "redux": "^3.5.2"
  }

```

node_modules/react-native-controllers/ios/RCCNavigationController.m:217:26: Implicit declaration of function 'objc_getAssociatedObject' is invalid in C99

node_modules/react-native-controllers/ios/RCCNavigationController.m:217:26: Declaration of 'objc_getAssociatedObject' must be imported from module 'ObjectiveC.runtime' before it is required

node_modules/react-native-controllers/ios/RCCNavigationController.m:217:26: Conflicting types for 'objc_getAssociatedObject'

node_modules/react-native-controllers/ios/RCCNavigationController.m:219:24: Implicit declaration of function 'objc_getAssociatedObject' is invalid in C99

node_modules/react-native-controllers/ios/RCCNavigationController.m:219:24: Declaration of 'objc_getAssociatedObject' must be imported from module 'ObjectiveC.runtime' before it is required

node_modules/react-native-controllers/ios/RCCNavigationController.m:219:24: Conflicting types for 'objc_getAssociatedObject'

node_modules/react-native-controllers/ios/RCCNavigationController.m:247:5: Implicit declaration of function 'objc_setAssociatedObject' is invalid in C99

node_modules/react-native-controllers/ios/RCCNavigationController.m:247:5: Declaration of 'objc_setAssociatedObject' must be imported from module 'ObjectiveC.runtime' before it is required

node_modules/react-native-controllers/ios/RCCNavigationController.m:247:5: Conflicting types for 'objc_setAssociatedObject'

node_modules/react-native-controllers/ios/RCCNavigationController.m:247:91: Declaration of 'OBJC_ASSOCIATION_RETAIN_NONATOMIC' must be imported from module 'ObjectiveC.runtime' before it is required

Full report

> Build target ReactNativeControllers of project ReactNativeControllers with configuration Debug
> 
> CompileC /Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/Objects-normal/x86_64/RCCNavigationController.o RCCNavigationController.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
>     cd /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios
>     export LANG=en_US.US-ASCII
>     export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
>     /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu99 -fobjc-arc -fmodules -fmodules-cache-path=/Users/nenuphar/Library/Developer/Xcode/DerivedData/ModuleCache -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/nenuphar/Library/Developer/Xcode/DerivedData/ModuleCache/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wno-deprecated-implementations -DDEBUG=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -mios-simulator-version-min=8.0 -g -Wno-sign-conversion -fobjc-abi-version=2 -fobjc-legacy-dispatch -iquote /Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/ReactNativeControllers-generated-files.hmap -I/Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/ReactNativeControllers-own-target-headers.hmap -I/Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/ReactNativeControllers-all-target-headers.hmap -iquote /Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/ReactNativeControllers-project-headers.hmap -I/Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Products/Debug-iphonesimulator/include -I../../react-native/React -I../../react-native/React/Base -I../../react-native/React/CSSLayout -I../../react-native/React/Executors -I../../react-native/React/Modules -I../../react-native/React/Profiler -I../../react-native/React/Views -I/Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/DerivedSources/x86_64 -I/Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/DerivedSources -F/Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Products/Debug-iphonesimulator -MMD -MT dependencies -MF /Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/Objects-normal/x86_64/RCCNavigationController.d --serialize-diagnostics /Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/Objects-normal/x86_64/RCCNavigationController.dia -c /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m -o /Users/nenuphar/Library/Developer/Xcode/DerivedData/myApp-beimalkorgslxfhikhdmdpxakczz/Build/Intermediates/ReactNativeControllers.build/Debug-iphonesimulator/ReactNativeControllers.build/Objects-normal/x86_64/RCCNavigationController.o
> 
> /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:206:10: warning: unused variable 'isHiddenBool' [-Wunused-variable]
>     BOOL isHiddenBool = setHidden ? [setHidden boolValue] : NO;
>          ^
> /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:217:26: error: implicit declaration of function 'objc_getAssociatedObject' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
>   NSString *callbackId = objc_getAssociatedObject(barButtonItem, &CALLBACK_ASSOCIATED_KEY);
>                          ^
> /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:217:26: error: declaration of 'objc_getAssociatedObject' must be imported from module 'ObjectiveC.runtime' before it is required
> In module 'Foundation' imported from ../../react-native/React/Base/RCTBridgeModule.h:10:
> In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
> In module 'Dispatch' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:15:
> In module 'os' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/dispatch/dispatch.h:48:
> In module 'ObjectiveC' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/os/object.h:75:
> /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/objc/runtime.h:1527:16: note: previous declaration is here
> OBJC_EXPORT id objc_getAssociatedObject(id object, const void *key)
>                ^
> /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:217:26: error: conflicting types for 'objc_getAssociatedObject'
>   NSString *callbackId = objc_getAssociatedObject(barButtonItem, &CALLBACK_ASSOCIATED_KEY);
>                          ^
> In module 'Foundation' imported from ../../react-native/React/Base/RCTBridgeModule.h:10:
> In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
> In module 'Dispatch' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:15:
> In module 'os' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/dispatch/dispatch.h:48:
> In module 'ObjectiveC' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/os/object.h:75:
> /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/objc/runtime.h:1527:16: note: previous declaration is here
> OBJC_EXPORT id objc_getAssociatedObject(id object, const void *key)
>                ^
> /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:219:24: error: implicit declaration of function 'objc_getAssociatedObject' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
>   NSString *buttonId = objc_getAssociatedObject(barButtonItem, &CALLBACK_ASSOCIATED_ID);
>                        ^
> /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:219:24: error: declaration of 'objc_getAssociatedObject' must be imported from module 'ObjectiveC.runtime' before it is required
> In module 'Foundation' imported from ../../react-native/React/Base/RCTBridgeModule.h:10:
> In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
> In module 'Dispatch' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:15:
> In module 'os' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/dispatch/dispatch.h:48:
> In module 'ObjectiveC' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/os/object.h:75:
> /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/objc/runtime.h:1527:16: note: previous declaration is here
> OBJC_EXPORT id objc_getAssociatedObject(id object, const void *key)
>                ^
> /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:219:24: error: conflicting types for 'objc_getAssociatedObject'
>   NSString *buttonId = objc_getAssociatedObject(barButtonItem, &CALLBACK_ASSOCIATED_ID);
>                        ^
> In module 'Foundation' imported from ../../react-native/React/Base/RCTBridgeModule.h:10:
> In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
> In module 'Dispatch' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:15:
> In module 'os' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/dispatch/dispatch.h:48:
> In module 'ObjectiveC' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/os/object.h:75:
> /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/objc/runtime.h:1527:16: note: previous declaration is here
> OBJC_EXPORT id objc_getAssociatedObject(id object, const void *key)
>                ^
> /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:220:60: warning: 'sendAppEventWithName:body:' is deprecated: Subclass RCTEventEmitter instead [-Wdeprecated-declarations]
>   [[[RCCManager sharedInstance] getBridge].eventDispatcher sendAppEventWithName:callbackId body:@
>                                                            ^
> In file included from /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:4:
> ../../react-native/React/Base/RCTEventDispatcher.h:64:1: note: 'sendAppEventWithName:body:' has been explicitly marked deprecated here
> - (void)sendAppEventWithName:(NSString _)name body:(id)body
>   ^
>   /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:247:5: error: implicit declaration of function 'objc_setAssociatedObject' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
>   objc_setAssociatedObject(barButtonItem, &CALLBACK_ASSOCIATED_KEY, button[@"onPress"], OBJC_ASSOCIATION_RETAIN_NONATOMIC);
>   ^
>   /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:247:5: error: declaration of 'objc_setAssociatedObject' must be imported from module 'ObjectiveC.runtime' before it is required
>   In module 'Foundation' imported from ../../react-native/React/Base/RCTBridgeModule.h:10:
>   In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
>   In module 'Dispatch' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:15:
>   In module 'os' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/dispatch/dispatch.h:48:
>   In module 'ObjectiveC' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/os/object.h:75:
>   /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/objc/runtime.h:1514:18: note: previous declaration is here
>   OBJC_EXPORT void objc_setAssociatedObject(id object, const void *key, id value, objc_AssociationPolicy policy)
>                ^
>   /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:247:5: error: conflicting types for 'objc_setAssociatedObject'
>   objc_setAssociatedObject(barButtonItem, &CALLBACK_ASSOCIATED_KEY, button[@"onPress"], OBJC_ASSOCIATION_RETAIN_NONATOMIC);
>   ^
>   In module 'Foundation' imported from ../../react-native/React/Base/RCTBridgeModule.h:10:
>   In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
>   In module 'Dispatch' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:15:
>   In module 'os' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/dispatch/dispatch.h:48:
>   In module 'ObjectiveC' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/os/object.h:75:
>   /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/objc/runtime.h:1514:18: note: previous declaration is here
>   OBJC_EXPORT void objc_setAssociatedObject(id object, const void *key, id value, objc_AssociationPolicy policy)
>                ^
>   /Users/nenuphar/Documents/projects/reactNative/myApp/node_modules/react-native-controllers/ios/RCCNavigationController.m:247:91: error: declaration of 'OBJC_ASSOCIATION_RETAIN_NONATOMIC' must be imported from module 'ObjectiveC.runtime' before it is required
>   objc_setAssociatedObject(barButtonItem, &CALLBACK_ASSOCIATED_KEY, button[@"onPress"], OBJC_ASSOCIATION_RETAIN_NONATOMIC);
>                                                                                         ^
>   In module 'Foundation' imported from ../../react-native/React/Base/RCTBridgeModule.h:10:
>   In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
>   In module 'Dispatch' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:15:
>   In module 'os' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/dispatch/dispatch.h:48:
>   In module 'ObjectiveC' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/os/object.h:75:
>   /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk/usr/include/objc/runtime.h:1493:5: note: previous declaration is here
>   OBJC_ASSOCIATION_RETAIN_NONATOMIC = 1, /_*< Specifies a strong reference to the associated object. 
>   ^
>   2 warnings and 10 errors generated.
> 
> what I'm I doing wrong here?

**Comment by SergeyKorochansky** [comment]
@folivi https://github.com/techery/react-native-controllers/commit/7e363e8ac129ec94289f98a227d521c4055b299a

**Comment by folivi** [comment]
Thanks @webzepter 
is there any other breaking change?

I'm getting this error now with a deadly simple code

`app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Expect all root views to have unique tag. Added 1 twice'
`

**Comment by SergeyKorochansky** [comment]
@folivi We use customised fork https://github.com/techery/react-native-controllers, but you can cherry-pick single commit as workaround. I haven't seen `NSInternalInconsistencyException` exception, maybe because we hadn't synchronise with upstream

**Comment by simonmitchell** [comment]
This is literally a one line import, how did you guys not catch this!? :P

Also get the `NSInternalInconsistencyException`, when using startTabBasedApp from react-native-navigation

**Comment by miladebadi** [comment]
For fixing NSInternalInconsistencyException, I made it to work by changing RCTRootView.m file located in React ios library -> Base-> RCTRootView.m. 
change:

``` objective-c
- (NSNumber *)reactTag
{
  RCTAssertMainQueue();
  if (!super.reactTag) {
    /**
     * Every root view that is created must have a unique react tag.
     * Numbering of these tags goes from 1, 11, 21, 31, etc
     *
     * NOTE: Since the bridge persists, the RootViews might be reused, so the
     * react tag must be re-assigned every time a new UIManager is created.
     */
    self.reactTag = [_bridge.uiManager allocateRootTag];
  }
  return super.reactTag;
}
```

to

``` objective-c
- (NSNumber *)reactTag
{
  RCTAssertMainQueue();
  return self.reactTag = [_bridge.uiManager allocateRootTag];
}
```

you can temporarily use this hack but I think it is not a good solution. maybe it gives some hints for fixing this issue.

for more info see this commit:
[https://github.com/facebook/react-native/commit/313d84d507689a78a9f4084b162ac87c09d7d5e3#diff-9bdd6391cd866b37f84e7e9b5f276ff9](https://github.com/facebook/react-native/commit/313d84d507689a78a9f4084b162ac87c09d7d5e3#diff-9bdd6391cd866b37f84e7e9b5f276ff9)

**Comment by aksonov** [comment]
So 0.32-rc fixes it, or not? Anyway, you could just set reactTag within RCCViewController, without patching, right?

**Comment by miladebadi** [comment]
I am quite sure they will not fix this in react native, because this problem is not related to react native itself. I tried to set reactTag within RCCViewController but no luck. reactTag is generated by RCTView at runtime, to fix this we should rewrite RCTView as a helper to give us access to reacttag. I think we should stick to my fix until the official release of react-native-navigation 2.0 which will be compatible with latest version of react-native.

**Comment by aksonov** [comment]
@knot123 Why just don't inherit RCTRootView and create our class instead of RCTRootView within RCCViewController?

**Comment by DanielZlotin** [comment]
@aksonov it's a possibility, but rather a last resort.. Let's try a few different things before that (as @simonmitchell said it may be fixed by adding an import)

**Comment by artald** [comment]
Fixed in version 2.1.0

It would be great if everyone that had a problem with it could verify that this fix works for them. thanks.
